kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/ErrorCollector.kt

```diff
@@ -29,38 +29,38 @@ interface ErrorCollector {
     */
    fun clear()
 
-   fun pushClue(clue: Any)
+   fun pushClue(clue: () -> String)
```

Comment:
I think we could probably turn ErrorCollector into an internal class.

Reply:
Same with the interface?